### PR TITLE
[kube-prometheus-stack-operator] add securityContext on container level for operator

### PR DESCRIPTION
**Pull request by ChrisFraun**
This PR is only a small change in the helm chart of **prometheus-operator**.

Signed-off-by: Christoph Fraundorfer <christoph.fraundorfer@allianzdirect.de>

#### What this PR does / why we need it:
**What**: Deployments can have security settings in their manifest on two levels: pod and container. However, there are some capabilities only configurable in one of the respective levels(https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#securitycontext-v1-core). This PR sets a default configuration for container securityContext, which denies privilege escalation and drops all POSIX capabilities. These are and should be standard settings in the context of Kubernetes. It also adds the possibility of running prometheus-operator in an Kubernetes environment without PSP (to be removed in v1.25 https://kubernetes.io/docs/concepts/security/pod-security-policy/), but with OpenPolicyAgent (a or possibly the PSP substitute) with the same capabilities as a restricted PSP instead.

#### Which issue this PR fixes
  - fixes #<4545> https://github.com/prometheus-operator/prometheus-operator/issues/4545

#### Special notes for your reviewer:
**Problem**: There is already a securityContext in the values.yaml, however, this should have entries for pods AND container be be called podSecurityContext since it's on pod level, but it hasn't. To not break backwards compatibility of prometheus-operator, this PR hardcodes the respective capabilities on container level with a new variable calles `containersSecurityContext`.

Other suggestions are welcome!

**Test**: was `helm lint .` and `helm template .`, which both compiled.

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
please raise minor chart version

**Comment by ChrisFraun** [comment]
Done!

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 39.1.0
+version: 39.2.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]:
```suggestion
version: 39.3.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by ChrisFraun** [comment]
I just realized that the future is already implemented.
https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L1832-L1834
https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml#L119

**Comment by ChrisFraun** [comment]
I just realized that the future is already implemented.
https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L1832-L1834
https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml#L119

**Pull request closed without merging**